### DISCUSSION TITLE
Reflecting Sybase ASE tables into MS SQL (Azure DB) - Syntax error in DDL

### DISCUSSION THREAD

**Discussion by ZachMassia** [sqlalchemy/sqlalchemy#9243] [Usage Questions] [answered]
Hi, 

I'm attempting to load metadata from Sybase ASE, and then run `create_all()` on an SQL Server connection, however the generated DDL is causing syntax errors.

It fails on the first column, because `create default zero_dflt as 0` or a variation of it, based on the datatype, is being appended to each line.

Full column line for example: `colname SMALLINT NOT NULL DEFAULT create default zero_dflt as 0,`

Here is a gist containing the relevant code, the full DDL, and the full error message: https://gist.github.com/ZachMassia/6bae47330d21e9da490178ea6c8f4f46

Thanks in advance if anyone can point me in the right direction!

**Comment by zzzeek** [comment, marked as answer]
we dont have a dialect for Sybase ASE here, and the one we used to have IIRC didn't actually reflect DB tables.

for an operation like this you likely want to get a raw DDL dump of your Sybase ASE database, then edit it directly using regular expression , search-and-replace etc. and then run the DDL script directly into SQL Server.

**Reply by ZachMassia** [reply]
Hi,

So here's the library I used to do the reflection: https://github.com/gordthompson/sqlalchemy-sybase

Seems to have picked everything up OK type wise (other than the few custom types I overrode). It's only when I go to write into the SQL azure DB that the generated DDL seems wrong. 

Perhaps I'll have to take this up on the sqlalchemy-sybase package. There is also a Microsoft migration program that I may be able to use if this won't work for the initial table creation. 

Once I get the source tables created, do you envision any similar issues just moving data from Sybase ASE -> SQL Server?

**Reply by zzzeek** [reply]
yeah @gordthompson  works with us but yeah he can handle that over in that repo (I forgot he took it over)

**Reply by ZachMassia** [reply]
Cool, I'll take it over to the other repo. Thanks!
